source/docs/software/hardware-apis/misc/addressable-leds.rst

```diff
@@ -107,50 +186,49 @@ LEDs can be set with the ``setHSV`` method that takes 4 arguments: index of the
-
-.. note:: It's good robot practice to keep the ``robotPeriodic()`` method as clean as possible, so we'll create a method for handling setting our LED data. We'll call this method ``rainbow()`` and call it from ``robotPeriodic()``.
+Using the built in ``LEDPattern.rainbow`` method, we can create a pattern that displays a full rainbow across an entire LED strip. Then, by calling ``scrollAtAbsoluteSpeed`` we can make it animate and cycle around the strip. ``rainbow`` accepts two arguments - one for the saturation and one for the value, expressed as a number from 0 to 255.
 
 .. tab-set::
 
    .. tab-item:: Java
       :sync: Java
 
-      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/v2024.3.2/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/addressableled/Robot.java
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/main/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/addressableled/Robot.java
          :language: java
-         :lines: 42-55
+         :lines: 21-31
          :linenos:
-         :lineno-start: 42
+         :lineno-start: 21
 
    .. tab-item:: C++
       :sync: C++
 
-      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/v2024.3.2/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/main/wpilibcExamples/src/main/cpp/examples/AddressableLED/include/Robot.h
          :language: c++
-         :lines: 22-35
+         :lines: 27-37
          :linenos:
-         :lineno-start: 22
+         :lineno-start: 27
 
-Now that we have our ``rainbow`` method created, we have to actually call the method and set the data of the LED.
+Now that the rainbow pattern is defined, we only need to apply it.
 
 .. tab-set::
 
    .. tab-item:: Java
       :sync: Java
 
-      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/v2024.3.2/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/addressableled/Robot.java
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/main/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/addressableled/Robot.java
          :language: java
-         :lines: 34-40
+         :lines: 50-56
          :linenos:
-         :lineno-start: 34
+         :lineno-start: 50
 
    .. tab-item:: C++
       :sync: C++
 
-      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/v2024.3.2/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/main/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp
          :language: c++
          :lines: 15-20
          :linenos:
          :lineno-start: 15
+
+.. image:: images/rainbow.gif
```

Comment:
I wonder if specifying the :width: will fix the pdf error

Reply:
🤔 Hmm. I'll need to test that. I don't have LaTeX tools installed locally so it'll need to be on CI, which could be frustrating.

The style guide _does_ explicitly call out gifs as being *un*acceptable for storage and accessibility concerns, but I don't see a way around it since the whole point of these docs are to demonstrate animations